docs/startup.md

```diff
@@ -0,0 +1,30 @@
+# Metrictank startup
+
+The full startup procedure has many details, but here we cover the main steps if they affect:
+
+* performance/resource usage characteristics
+* cluster status
+* API availability
+* diagnostics
+
+
+| Phase                   | Description                                                                                        | effect on CPU / RAM                 |
+| ----------------------- | -------------------------------------------------------------------------------------------------- | ----------------------------------- |
+| load config             | load/validate config                                                                               | no                                  |
+| setup diagnostics       | set up logging, profiling, proftrigger                                                             | no                                  |
+| log startup             | logs "Metrictank starting" message                                                                 | no                                  |
+| start sending stats     | starts connecting and writing to graphite endpoint                                                 | no                                  |
+| create Store            | create keyspace, tables, write queues, etc                                                         | minor RAM increase ~ queue size     |
+| create Input(s)         | open connections (kafka) or listening sockets (carbon, prometheus)                                 | no                                  |
+| start cluster           | starts gossip, joins cluster                                                                       | no                                  |
+| create Index            | creates instance and starts write queues                                                           | minor RAM increase ~ queue size     |
+| start API server        | opens listening socket and starts handling requests                                                | no                                  |
```

Comment:
This is a little confusing because it sounds like MT will already handle queries, but it loads a lot of data after this step. Maybe it should be mentioned that even though the sockets are listening, it will reject queries at this point until it is `ready`.